src/golang/lib/k8s/clients.go

```diff
@@ -0,0 +1,85 @@
+package k8s
+
+import (
+	"context"
+
+	"github.com/dropbox/godropbox/errors"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// This key is used to create a label for every pod that gives the name of
+	// the service that it is running. For user workloads, the value will be
+	// the user-specified service name.
+	ServiceKey = "serviceName"
+)
+
+// This is a helper function that creates a Kubernetes Clientset using the
+// `client-go/rest` library's `InClusterConfig()` function. This function will
+// only succeed if it is called from within a Kubernetes cluster. Otherwise,
+// the configuration retrieval will fail, and this function will call
+// `log.Fatal` and cause the program to crash. If this function is called from
+// within the cluster, it should not fail.
+func CreateClientInCluster() *kubernetes.Clientset {
+	k8sConfig, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Unexpected error while creating Kubernetes client: %v\n", err)
```

Comment:
Lets remove these `log.Fatal`s and return an actual error that the caller can use